ui/desktop/app/routes/scopes.js

```diff
@@ -23,9 +44,53 @@ export default class ScopesRoute extends Route {
    * @param {string} params.scope_id
    * @return {Promise{[ScopeModel]}}
    */
-  model() {
+  async model() {
     // NOTE:  In the absence of a `scope_id` query parameter, this endpoint is
     // expected to default to the global scope, thus returning org scopes.
     return this.store.query('scope', {}).catch(() => A([]));
   }
+
+  /**
+   * Adds `isPaginationSupported` to the controller.
+   * @param {Controller} controller
+   */
+  async setupController(controller) {
+    super.setupController(...arguments);
+
+    const isPaginationSupported = get(this, 'isPaginationSupported');
+    controller.set('isPaginationSupported', isPaginationSupported);
+
+    let downloadLink;
+    let downloadError = false;
+
+    if (!isPaginationSupported) {
+      const metaDataUrl =
+        'https://api.releases.hashicorp.com/v1/releases/boundary-desktop/1.7.1';
+      const { isWindows, isMac, isLinux } = await this.ipc.invoke('checkOS');
+
+      try {
+        const metaDataResponse = await fetch(metaDataUrl);
+        const metaData = await metaDataResponse.json();
+
+        if (isWindows) {
+          downloadLink = this.extractOsSpecificUrl(metaData, 'windows');
+        } else if (isMac) {
+          downloadLink = this.extractOsSpecificUrl(metaData, 'darwin');
+        } else if (isLinux) {
+          downloadLink = this.extractOsSpecificUrl(metaData, 'linux');
+        }
+      } catch (e) {
+        // this is a catch for any errors that may occur and shows the user
+        // an error alert directing them to the releases page
+        downloadError = true;
+      }
+    }
+
+    controller.set('downloadLink', downloadLink);
+    controller.set('downloadError', downloadError);
+  }
+
+  extractOsSpecificUrl(metaData, os) {
+    return metaData.builds.find((build) => build.os === os).url;
```

Comment:
Is the order guaranteed? How do we know if we're grabbing the `zip` or `dmg` for example

Reply:
Great question. I updated this to also grab the correct url by file extension. If it fails to find a url from the metaData, it will throw an error.